python_modules/dagster/dagster/_core/external_execution/context.py

```diff
@@ -1,21 +1,64 @@
-from typing import Optional
+from typing import Any, Optional
 
 from dagster_externals import (
     ExternalExecutionContextData,
     ExternalExecutionDataProvenance,
     ExternalExecutionExtras,
+    ExternalExecutionMessage,
     ExternalExecutionTimeWindow,
 )
 
-from dagster._core.definitions.data_version import DataProvenance
+import dagster._check as check
+from dagster._core.definitions.data_version import DataProvenance, DataVersion
 from dagster._core.definitions.events import AssetKey
 from dagster._core.definitions.partition_key_range import PartitionKeyRange
 from dagster._core.definitions.time_window_partitions import TimeWindow
 from dagster._core.execution.context.compute import OpExecutionContext
 from dagster._core.execution.context.invocation import BoundOpExecutionContext
 
 
-def build_external_execution_context(
+class ExternalExecutionOrchestrationContext:
```

Comment:
Re the naming here. Do you imagine the responsibilities of this class growing in the future? Why is this a context and not "message handler" or something like that.

Reply:
I think it is natural for this to be a "context" because:

- Currently this is consumed by _both_ "channel"-like entities in the orchestration process-- the `file_context_source` and `file_message_sink`. It both handles messages and provides the `ExternalContextExecutionData` that is written by the context source.
- It encapsulates same data as `ExternalExecutionContext` on the other end (info from `OpExecutionContext` + extras)
- It provides an interface over an existing context (`OpExecutionContext`), similar to `OpExecutionContext` relationship to `StepExecutionContext`

Reply:
Cool thanks for explanation